src/Storage/DBFile.php

```diff
@@ -359,12 +368,22 @@ public function getVisibility()
 
     public function exists()
     {
```

Comment:
hmm failed unit test in travis, disabling caching here means the test pass

May have to make caching configurable and disabled by default.  Then remove the calls  to $this->resetExistsCached(); 

Then turn on caching only when required.  

One example place to turn it one would be where 'ReadFiles' graphql request interacts with php - that request won't do any modifies or deletes so it should be safe
https://github.com/silverstripe/silverstripe-asset-admin/blob/1/code/GraphQL/ReadFileQueryCreator.php#L26